source/portable/os/ota_os_posix.c

```diff
@@ -117,27 +122,49 @@ OtaOsStatus_t Posix_OtaSendEvent( OtaEventContext_t * pEventCtx,
                                   unsigned int timeout )
 {
     OtaOsStatus_t otaOsStatus = OtaOsSuccess;
+    struct timespec ts = { 0 };
+    struct pollfd fds = { 0 };
+    int pollResult = 0;
 
     ( void ) pEventCtx;
-    ( void ) timeout;
-
-    /* Send the event to OTA event queue.*/
-    errno = 0;
 
-    if( mq_send( otaEventQueue, pEventMsg, MAX_MSG_SIZE, 0 ) == -1 )
+    if( timeout > INT32_MAX )
     {
         otaOsStatus = OtaOsEventQueueSendFailed;
 
-        LogError( ( "Failed to send event to OTA Event Queue: "
-                    "mq_send returned error: "
-                    "OtaOsStatus_t=%i "
-                    ",errno=%s",
-                    otaOsStatus,
-                    strerror( errno ) ) );
+        LogError( ( "Invalid input, Posix_OtaSendEvent supports timeout < INT32_MAX, "
+                    "timeout = %u",
+                    timeout ) );
     }
     else
     {
-        LogDebug( ( "OTA Event Sent." ) );
+        /* Send the event to OTA event queue.*/
+        errno = 0;
+
+        fds.fd = otaEventQueue;
+        fds.events = POLLOUT;
+
+        pollResult = poll( &fds, 1, ( int ) timeout );
```

Comment:
The poll / mq_send calls should be in a timeout loop in a multiple-producer, single-consumer situation, to account for a race condition when two threads are both blocked sending to the same queue.

Reply:
Thanks for suggestion! Added loop to handle race conditions.
BTW, add pollAndSend and pollAndReceive internal functions to reduce code complexity level.